Challenges/challenge_46_can_place_flowers/greghilston/src/main/java/Solution.java

```diff
@@ -0,0 +1,54 @@
+public class Solution  {
+    /**
+     * Determines if we can plant a flower at a given location in a specific flowerbed.
+     * @param flowerBed state of flowerbed
+     * @param i location to check
+     * @return whether we can plant a flower or not
+     */
+    private boolean canPlantHere(int[] flowerBed, int i) {
+        // left check
+        int leftNeighborIndex = i - 1;
+
+        if(leftNeighborIndex > 0) // ensures we don't go off the flower bed
+        {
+            if(flowerBed[leftNeighborIndex] == 1) {
```

Comment:
Since this is a very, very small and simple algorithm, we might as well talk nano-optimizations: Why have branching multiple times when you can just `&&` the conditions?

Reply:
Yea that's true. Changing this up now. I generally "think" in nested ifs, but I'm totally with you.